From ab0323450a4fca09eb8afd07b029836027234801 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@logicpd.com>
Date: Mon, 15 Jun 2020 19:27:09 -0500
Subject: [PATCH 30/30] dts: Add pin mapping for SCIF4 -> /dev/ttySC6

Disable DPI (rgb / parallel) display by default
Add -rgb dts file
---
 .../dts/renesas/beacon-renesom-baseboard.dtsi | 15 +++++++++++++--
 .../renesas/r8a774a1-beacon-rzg2m-kit-rgb.dts | 19 +++++++++++++++++++
 2 files changed, 32 insertions(+), 2 deletions(-)
 mode change 100644 => 100755 arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
 create mode 100644 arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit-rgb.dts

diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
old mode 100644
new mode 100755
index 82864ed7e655..8e1e6e1a7f2f
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
@@ -17,6 +17,7 @@
 		serial3 = &scif0;
 		serial4 = &hscif2;
 		serial5 = &scif5;
+		serial6 = &scif4;
 		spi0 = &msiof0;
 		spi1 = &msiof1;
 		spi2 = &msiof2;
@@ -311,8 +312,6 @@
 };
 
 &du {
-	pinctrl-0 = <&du_pins>;
-	pinctrl-names = "default";
 	status = "okay";
 
 	clocks = <&cpg CPG_MOD 724>,
@@ -578,6 +577,11 @@
 };
 
 &pfc {
+	scif4_pins: scif4 {
+		groups = "scif4_data_c";
+		function = "scif4";
+	};
+
 	can0_pins: can0 {
 		groups = "can0_data_a";
 		function = "can0";
@@ -755,6 +759,13 @@
 	status = "okay";
 };
 
+&scif4 {
+	pinctrl-0 = <&scif4_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
 &scif5 {
 	pinctrl-0 = <&scif5_pins>;
 	pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit-rgb.dts b/arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit-rgb.dts
new file mode 100644
index 000000000000..52d756258951
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit-rgb.dts
@@ -0,0 +1,19 @@
+/dts-v1/;
+
+#include "r8a774a1.dtsi"
+#include "beacon-renesom-som.dtsi"
+#include "beacon-renesom-baseboard.dtsi"
+
+/ {
+	model = "Beacon Embedded Works RZ/G2M Development Kit";
+	compatible =	"beacon,beacon-rzg2m", "renesas,r8a774a1";
+};
+
+&scif4 {
+	status = "disabled";
+};
+
+&du {
+	pinctrl-0 = <&du_pins>;
+	pinctrl-names = "default";
+};
-- 
2.17.1

